ts/query.go

```diff
@@ -358,6 +371,16 @@ func (is unionIterator) sumAvg() float64 {
 	return sum
 }
 
+// avg returns the sum of the differential for the averages of all iterators in
+// the set.
+func (is unionIterator) dAvg() float64 {
```

Comment:
It's actually kind of complicated to explain - there are two aggregations going on.  

One is the "downsampling" aggregation, which combines multiple points in the same series over a set time span. Downsampling first divides time into consecutive "sample periods" of a uniform length. Let there be two data points in the series at time "x" and time "y", where x and y fall within the same sample period; the downsampling aggregator will combine the values at time "x" and "y" into one value. 

The other is the "union" aggregation of values from multiple series into a single set of results - that is, if a query is returning data from two series "A" and "B", and both series have a data value at time "X", then the values for both series at time X will be aggregated using the "union" aggregator function.

At this point, the *union* aggregator is *always* a sum.  What we are exposing is the *downsampling* aggregator, which was previously always "average" but now can be "average_rate".